openbadge-server/openbadge/views.py

```diff
@@ -372,48 +396,72 @@ def post_datafile(request, project_key):
             "details": "No data provided!",
             "chunks_written": 0,
             "chunks_received": 0
-        })
+        }, status_code=400)
 
     chunks = request.data.get("chunks")
-    # I don't like this but it works for now
+
     # sometimes we don't get json objects from the request object
     # (with tests, but I don't know if it happens anywhere else?)
+    # we need it in json instead of a list of strings so we can sort/operate on it
     if not isinstance(chunks, dict) and not isinstance(chunks, list):
         chunks = simplejson.loads(chunks)
 
+    # we sort the incoming data to be able to properly group the data
+    chunks = sorted(chunks, key=lambda chunk: chunk["data"]["timestamp"])
+
+    # now group by date for bulk writing to appropriate file
+    grouped_chunks = {}
+    for key, group in groupby(
+            chunks,
+            lambda chunk: datetime.fromtimestamp(chunk["data"]["timestamp"]).date().strftime('%Y-%m-%d')):
```

Comment:
Is this function using local time (of the server) or UTC time? We'll probably want local time here